adagrams/game.py

```diff
@@ -1,13 +1,197 @@
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # first need to understand what are letters in word and how many of the letter do we have.
+    # then I need to compare each letter with my draw bank:
+    # if the letter in the word doesn't exist in the draw return false
+    # if the letter is in the draw bank then check if the frequency in the word 
+    # is more than the frequency of same letter in the draw bank
+    
+    word = word.upper()
+    word_dict = make_dict(word)
+    letter_bank_dict = make_dict(letter_bank)
+
+    for character, frequency in word_dict.items():
+        if  character not in letter_bank_dict:
+            return False
+        else:
+            if frequency > letter_bank_dict[character]:
+                return False
+    return True
+    
 
 def score_word(word):
-    pass
+    # compare the letters in the word with the letters in score chart. 
+    # add corresponding value to the total score
+    # Chech if the length of word is more than 7 
+    # (should we check if it is more than 10?)
+    # return the total score
+
+
+    # letter_bank = draw_letters() 
+    word = word.upper()
+    word_dict = make_dict(word)
+
+    # if uses_available_letters(word, letter_bank): # No need to check!
+    total_score = 0
+
+    for letter, frequency in word_dict.items():
+        total_score += score_chart[letter] * frequency
+
+    if len(word) >= 7:
+        total_score += 8
+
+    return total_score
+    
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    # we need to create a dictionary of words as keys and scores as values. (should we add the word length here?)
+    # start the comparison: iterating over the dictionary and compare the values to get the highest score
+    # within this loop we should consider the tie cases: with if statement, check the score is greater, or =
+    # in case of equal scores: check the length of the words > smaller word is the winner!
+    # BUT if length of the word is 10, and there are multiple tie score, 10 letter word is the winner!
+    # if score and length are the same pich the first one as winner!
+    
+    highest_score = 0
+    winner_word = None
+
+    for word in word_list:
+        score = score_word(word)
+
+        if score > highest_score:
+            winner_word = word
+            highest_score = score
+     
+        elif score == highest_score:
+            if len(word) == 10 and len(winner_word) != 10:
+                winner_word = word              
+            elif len(word) < len(winner_word) and len (winner_word) != 10:
+                winner_word = word
+            else:
+                continue
+
+    return (winner_word, highest_score)
+
+# best_word = get_highest_word_score(["BBBBBB", "AAAAAAAAAA"])
+# print(best_word[0])
+# print(best_word[1])
+
+# best_word = get_highest_word_score(["JQ", "FHQ", "AAAAAAAAAA", "BBBBBB", "TTTTTTTTTT"])
+# print(best_word[0])
+# print(best_word[1])
```

Comment:
When submitting production level code, just make sure to remove unnecessary comments!